sickbeard/providers/btdigg.py

```diff
@@ -0,0 +1,176 @@
+                        search_string.append(ep_string)                       
+        return search_string
+
+
+    def _get_episode_search_strings(self, ep_obj):
+        
+        search_string = []
+       
+        if not ep_obj:
+            return []
+        if ep_obj.show.air_by_date:
+            for show_name in set(show_name_helpers.allPossibleShowNames(ep_obj.show)):
+                ep_string = show_name_helpers.sanitizeSceneName(show_name) +' '+ str(ep_obj.airdate).replace('-', '.')
+                search_string.append(ep_string)
+        else:
+            for show_name in set(show_name_helpers.allPossibleShowNames(ep_obj.show)):
+                ep_string = show_name_helpers.sanitizeSceneName(show_name) +' '+ sickbeard.config.naming_ep_type[2] % {'seasonnumber': ep_obj.season, 'episodenumber': ep_obj.episode}
+                search_string.append(ep_string)
+        return search_string    
+
+    def _doSearch(self, search_params, search_mode='eponly', epcount=0, age=0, epObj=None):
+        search_params = search_params.replace('+',' ')
+        logger.log("Performing Search: {0}".format(search_params))
+        searchUrl = self.url + "api/private-341ada3245790954/s02?q=" + search_params + "&p=0&order=0"
+        return self.parseResults(searchUrl)
+    
+
+    def _get_title_and_url(self, item):
+        return item
+    
+    def parseResults(self, searchUrl):
+        data = self.getURL(searchUrl)
+        results=[]
+        tmp_results=[]
+        if data:
+            logger.log("parseResults() URL: " + searchUrl, logger.DEBUG)
+            jdata = json.load(data)
+            for torrent in sorted(jdata, key=itemgetter('reqs'), reverse=True):
+                found=0
+                
+                if torrent['ff'] > 0.0:
+                    continue
+                               
+                torrent['name'] = torrent['name'].replace('|', '').replace('.',' ')
+                
+                item = (torrent['name'],torrent['magnet'], torrent['reqs'])                
+                for r in tmp_results:
+                    if r[0].lower() == torrent['name'].lower():
+                        found=1
+                        if r[2] < torrent['reqs']:
+                            tmp_results[tmp_results.index(r)]=item
+                            
+                if not found:
+                    tmp_results.append(item)
+                logger.log("parseResults() Title: " + torrent['name'], logger.DEBUG)
+            
+            for r in tmp_results:
+                item=(r[0],r[1])
+                results.append(item)
+        else:
+            logger.log("parseResults() Error no data returned!!")
+        return results
+
+    
+    def getURL(self, url, headers=None):
+        response = None
+        logger.log("Requesting - " + url)
+        try:
+            #response = helpers.getURL(url, headers)
+            response = urllib.urlopen(url)
```

Comment:
Can't use this as it. Add  `self.session = requests.Session()` in `__init__` and just remove this `getURL` function altogether.